airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/AirbyteMessageTracker.java

```diff
@@ -368,6 +406,15 @@ public Map<AirbyteStreamNameNamespacePair, Long> getStreamToEmittedRecords() {
         entry -> nameNamespacePairToIndex.inverse().get(entry.getKey()), Entry::getValue));
   }
 
+  /**
+   * Swap out stream indices for stream names and return total records estimated by stream.
+   */
+  @Override
+  public Map<AirbyteStreamNameNamespacePair, Long> getStreamToEstimatedRecords() {
```

Comment:
Looking at the PR, this feels like a lot of duplication. Does it make sense to change the storage to a `Map<AirbyteStreamNameNamespacePair, SyncStats>` where SyncStats would be a POJO of all the different stats we may track?

Reply:
I thought of that and decided to punt it since plenty of stuff is changing in this PR.

The fact that you brought it up makes me think it's worth doing now. Will do so and update you.

Reply:
@gosusnp I added a StreamStats POJO. This is currently on in use for the estimated stats. I plan to refactor the rest of the stats in a follow up PR since the changeset does get big. How is this for now?

Reply:
to allay your fears - the follow up PR should be out next day or so, so we shouldn't stay in this messy state for long.

Reply:
Looks better. I think we can simplify this a bit more to avoid having different loops, this can be tackled later.

What I am thinking: since I would expect that most of the time, we want all the stats for a given stream, I think we be better off just getting the sum of all the stats rather than have one function that returns one map per stat.

Reply:
I like it!

Reply:
will do so in a follow up PR